src/main/java/net/sf/marineapi/nmea/io/DefaultDataReader.java

```diff
@@ -52,6 +52,10 @@ public DefaultDataReader(InputStream source, SentenceReader parent) {
 	 */
 	@Override
 	public String read() throws Exception {
-		return input.readLine();
+		final String result = input.readLine();
+		if (result == null) {
+		    setRunning(false);

```

Comment:
Hmm, true. I based my comment on recalling that previously it was possible to paste more data in a file and the reader continued from that. Now it doesn't seem to work like that anymore, which also makes pause events quite useless if it never continues. This is likely due to changes made in reader in v0.9, will have to check this.